singleton-service/src/service/proxy.rs

```diff
@@ -0,0 +1,155 @@
+pub use crate::configuration::service::ServiceConfig;
+pub use crate::service::deltas::DeltaStore;
+use log::{debug, info, warn};
+use proxy_wasm::{
+    hostcalls::{dequeue_shared_queue, register_shared_queue},
```

Comment:
Just a small nit: You can call host functions using `self.` inside implementation for any Context (Root or Http). You can remove these imports and use `self.host_call`.

Reply:
yeah but there's a slight difference in the APIs. The APIs in the hostcalls are wrapped with `Result`